### PR TITLE
Stories/dt/checknokeyword #89

### PR DESCRIPTION
**Pull request by terw-dan**
Changed Metadescription, Linkcount and imagecount to return results when no keyword is set. 

Changed the FleschReading so it doesn't calculate on empty textstring.

### CODE REVIEW

File: dist/yoast-seo.js [hunk 1]
```diff
@@ -264,30 +264,32 @@ YoastSEO.Analyzer.prototype.stopwords = function() {
  * @returns {result object}
  */
 YoastSEO.Analyzer.prototype.fleschReading = function() {
-	var score = (
+	if ( this.preProcessor.__store.wordcountNoTags > 0 ) {
+		var score = (
```
Comment (CarolineGeven) [inline dist/yoast-seo.js]: Please add the formula for the Flesch reading-ease score so we understand where the numbers are coming from

File: dist/yoast-seo.js [hunk 2]
```diff
@@ -334,7 +337,11 @@ YoastSEO.Analyzer.prototype.linkCount = function() {
 		for ( var i = 0; i < linkMatches.length; i++ ) {
 			var linkKeyword = this.linkKeyword( linkMatches[ i ] );
 			if ( linkKeyword ) {
-				linkCount.totalKeyword++;
+				if ( this.config.keyword !== "" ) {
```
Comment (CarolineGeven) [inline dist/yoast-seo.js]: Why do we need to do a seperate check for totalKeyword and totalNAKeyword? If we always want to know the total link count, we don't need to check if there's a keyword set or not.

File: dist/yoast-seo.js [hunk 3]
```diff
@@ -426,7 +433,7 @@ YoastSEO.Analyzer.prototype.linkResult = function( obj ) {
  * @returns {{name: string, result: {total: number, alt: number, noAlt: number}}}
```
Comment (CarolineGeven) [inline dist/yoast-seo.js]: Missing `altNaKeyword` in comments

File: dist/yoast-seo.js [hunk 4]
```diff
@@ -437,12 +444,15 @@ YoastSEO.Analyzer.prototype.imageCount = function() {
 			//matches everything in the alt attribute, case insensitive and global.
 			var alttag = imageMatches[ i ].match( /alt=([\'\"])(.*?)\1/ig );
 			if ( this.imageAlttag( alttag ) ) {
-				if ( this.imageAlttagKeyword( alttag ) ) {
-					imageCount.altKeyword++;
+				if ( this.config.keyword !== "" ) {
+					if ( this.imageAlttagKeyword( alttag ) ) {
+						imageCount.altKeyword++;
+					} else {
+						imageCount.alt++;
```
Comment (CarolineGeven) [inline dist/yoast-seo.js]: Is it safe to assume `alt` stores the amount of images without the focus keyword in the alt tag? Please add inline docs to clarify.

File: spec/metadescriptionKeywordSpec.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	it("returns a match for the keyword", function(){
+		var metaAnalyzer = Factory.buildAnalyzer(metaArgs);
+		var result = metaAnalyzer.metaDescriptionKeyword();
+		expect(result[0].result).toBe(1);
+	});
+});
+
+var metaArgs2 = {
+	keyword: "sample",
+	meta: "this is the metadescription that doesn't contains a keyword"
```
Comment (CarolineGeven) [inline spec/metadescriptionKeywordSpec.js]: should be `contain` ;)

File: spec/metadescriptionKeywordSpec.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	keyword: "sample",
+};
+
+describe("a test matching the keywords in the metadescription", function(){
+	it("returns no matches for the keyword, since there is no metadescription", function(){
+		var metaAnalyzer = Factory.buildAnalyzer(metaArgs3);
+		var result = metaAnalyzer.metaDescriptionKeyword();
+		expect(result[0].result).toBe(0);
+	});
+});
```
Comment (CarolineGeven) [inline spec/metadescriptionKeywordSpec.js]: Missing a test with no focus keyword and a meta description (and, for completion perhaps a test with no keyword and no meta description

**Comment by CarolineGeven** [comment]
CR1 done

File: dist/yoast-seo.js (hunk 2 above)
Comment (terw-dan) [inline dist/yoast-seo.js]: This is saved seperately so we can differentiate in the scoring, if no keyword is found the scoring is different.

**Comment by CarolineGeven** [comment]
CR2 done :+1:

**Pull request merged**